### PR TITLE
Display version of osmtm from last git commit

### LINKED ISSUE #458: Show the release version
The current deployed version should be visible in the interface. See iD editor as an example.
This should be as automatic as possible.

### PR DESCRIPTION
**Pull request by pyrog**
use package gitversion

fix #458

### CODE REVIEW

**Comment by pgiraud** [comment]
Your pull request cannot be merged. Please rebase it first.

**Comment by pgiraud** [comment]
Sorry to bother, but can you please use rebase instead of merge?

**Comment by pgiraud** [comment]
The automated tests fail unfortunately.
The `versionfromgit` package is not installed. Adding it as a submodule is not enough. It needs to be installed in the virtualenv.
Moreover, I don't think you need to load it as a submodule. This package is available on pypi so you should only need to add it to the list of dependencies in `setup.py`.

**Comment by pyrog** [comment]
```versionfromgit``` package was a "waste" from a previous test.

I clean my commit but it failed:
```
Command python setup.py egg_info failed with error code 1 in /home/travis/build/hotosm/osm-tasking-manager2
Storing debug log for failure in /home/travis/.pip/pip.log
The command "pip install -r requirements.txt" failed and exited with 1 during .
```

**Comment by ethan-nelson** [comment]
`ImportError: No module named version`

**Comment by pyrog** [comment]
Thanks Ethan :smile: 

nosetests pass on my server, but failed on github. @pgiraud could you check what's wrong with **requirements.txt** please ?

**Comment by pgiraud** [comment]
The problem is that you're importing the `gitversion` in the setup.py. When one wants to install the application, `setup.py`is actually executed. But at that time `gitversion` package has not been installed yet.

**Comment by pgiraud** [comment]
In other words, the package needs to be installed prior to installing the application.
https://github.com/mikebryant/python-gitversion/issues/1

By the way, I don't think we really need to care about the version number in `setup.py`. I think it's the code manager to take care of this when releasing.
https://github.com/hotosm/osm-tasking-manager2/blob/master/RELEASING.md

**Comment by pyrog** [comment]
So, what should I do ?

Remove in setup.py #L4
```python
from osmtm.version import __VERSION__
```
and keep #L42
```
      version='2.9-dev',
```
and of course, display version number in about.mako ?

**Comment by pgiraud** [comment]
I'm probably a bit picky but I don't think we need to have a specific python file to manage the version. Everything could be done in the mako template now.

The git hook may be an option.

**Comment by pyrog** [comment]
Ok Boss :wink: 
I'll put theses lines in about.mako :smile:
```python
<%
    from gitversion import determine_git_version
%>
      <h3>Version</h3>
      <p>
        ${determine_git_version(__file__)}
      </p>
```

**Comment by pyrog** [comment]
Ok Boss :wink: 
I'll put theses lines in about.mako :smile:
```python
<%
from gitversion import determine_git_version
%>
…
${determine_git_version(__file__)}
```

File: osmtm/version.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from gitversion import determine_git_version
+__VERSION__ = determine_git_version(__file__)
```
Comment (pgiraud) [inline osmtm/version.py]: I don't think we need this file anymore.

Comment (pgiraud) [inline osmtm/version.py]: @pyrog can you please remove the `version.py` file. I'd be happy to merge it then. Thanks.

**Pull request merged**